hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/AvroConversionUtils.scala

```diff
@@ -62,10 +62,10 @@ object AvroConversionUtils {
    * @param rootCatalystType Catalyst [[StructType]] to be transformed into
    * @return converter accepting Avro payload and transforming it into a Catalyst one (in the form of [[InternalRow]])
    */
-  def createAvroToInternalRowConverter(rootAvroType: Schema, rootCatalystType: StructType): GenericRecord => Option[InternalRow] =
-    record => sparkAdapter.createAvroDeserializer(rootAvroType, rootCatalystType)
```

Comment:
@xiarixiaoyao that's a very sneaky issue

Please leave a note in a comment explaining what was the issue and how you've addressed it.

However, i'd suggest to just pull out SparkAdapter from the closure keeping the API of this method intact: we don't want to push casting (to InternalRow/GenericRecord) onto the users

```
def createAvroToInternalRowConverter(rootAvroType: Schema, rootCatalystType: StructType): GenericRecord => Option[InternalRow] = {
    val deserilizer = sparkAdapter.createAvroDeserializer(rootAvroType, rootCatalystType)
    record => deserializer.deserialize(record).map(_.asInstanceOf[InternalRow])
}
```

Reply:
fixed